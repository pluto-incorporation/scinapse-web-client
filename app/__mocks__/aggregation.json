{
  "data": {
    "years": [
      { "year": 2008, "doc_count": 0 },
      { "year": 2009, "doc_count": 25 },
      { "year": 2010, "doc_count": 14 },
      { "year": 2011, "doc_count": 13 },
      { "year": 2012, "doc_count": 13 },
      { "year": 2013, "doc_count": 12 },
      { "year": 2014, "doc_count": 13 },
      { "year": 2015, "doc_count": 0 },
      { "year": 2016, "doc_count": 0 },
      { "year": 2017, "doc_count": 0 },
      { "year": 2018, "doc_count": 0 }
    ],
    "journals": [
      { "id": 25293849, "title": "The FASEB Journal", "doc_count": 2 },
      { "id": 107775873, "title": "Hydrobiologia", "doc_count": 2 },
      { "id": 3880285, "title": "Science", "doc_count": 1 },
      { "id": 11380747, "title": "Canadian Veterinary Journal-revue Veterinaire Canadienne", "doc_count": 1 },
      { "id": 11973042, "title": "Plant Archives", "doc_count": 1 },
      { "id": 12065667, "title": "Mikrochimica Acta", "doc_count": 1 },
      { "id": 13275601, "title": "American Scholar", "doc_count": 1 },
      { "id": 20845397, "title": "Journal of The Canadian Dental Association", "doc_count": 1 },
      { "id": 24807848, "title": "Physical Review Letters", "doc_count": 1 },
      { "id": 26277462, "title": "International Journal of Agriculture and Biology", "doc_count": 1 }
    ],
    "impact_factors": [
      { "from": 0, "to": 1, "doc_count": 9 },
      { "from": 1, "to": 2, "doc_count": 9 },
      { "from": 2, "to": 3, "doc_count": 7 },
      { "from": 3, "to": 4, "doc_count": 2 },
      { "from": 4, "to": 5, "doc_count": 2 },
      { "from": 5, "to": 6, "doc_count": 2 },
      { "from": 6, "to": 7, "doc_count": 0 },
      { "from": 7, "to": 8, "doc_count": 0 },
      { "from": 8, "to": 9, "doc_count": 1 },
      { "from": 9, "to": 10, "doc_count": 0 },
      { "from": 10, "to": null, "doc_count": 32 }
    ],
    "fos_list": [
      { "id": 127413603, "name": "Engineering", "doc_count": 33 },
      { "id": 71924100, "name": "Medicine", "doc_count": 29 },
      { "id": 185592680, "name": "Chemistry", "doc_count": 24 },
      { "id": 41008148, "name": "Computer Science", "doc_count": 22 },
      { "id": 121332964, "name": "Physics", "doc_count": 21 },
      { "id": 86803240, "name": "Biology", "doc_count": 20 },
      { "id": 192562407, "name": "Materials Science", "doc_count": 14 },
      { "id": 17744445, "name": "Political Science", "doc_count": 13 },
      { "id": 33923547, "name": "Mathematics", "doc_count": 12 },
      { "id": 46141821, "name": "Nuclear magnetic resonance", "doc_count": 9 }
    ]
  }
}
