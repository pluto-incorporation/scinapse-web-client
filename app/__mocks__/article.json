{
  "id": 20,
  "type": "POST_PAPER",
  "title": "this is title",
  "articleAbstract": "this is abstract.",
  "summary": "this is summary.",
  "link": "this is summary.",
  "source": "ARXIV",
  "articlePublishedAt": "2017-10-10T10:10:10",
  "articleUpdatedAt": "2017-10-10T10:10:10",
  "point": {
    "total": 7,
    "originality": 7,
    "contribution": 7,
    "analysis": 7,
    "expressiveness": 7
  },
  "createdBy": {
    "id": 10,
    "email": "test",
    "name": "tedyeee",
    "reputation": 16
  },
  "createdAt": "2017-10-09T10:47:32.105",
  "authors": [
    {
      "id": 30,
      "type": "LEAD_AUTHOR",
      "name": "pepsi",
      "organization": "Coca Cola"
    },
    {
      "id": 31,
      "member": {
        "id": 10,
        "email": "test",
        "name": "tedyeee",
        "reputation": 16
      },
      "type": "CORRESPONDING_AUTHOR",
      "name": "coke",
      "organization": "kingsi"
    }
  ],
  "evaluations": [
    {
      "id": 50,
      "point": {
        "total": 7,
        "originality": 7,
        "contribution": 7,
        "analysis": 7,
        "expressiveness": 7,
        "originalityComment": "originality comment",
        "contributionComment": "contribution comment",
        "analysisComment": "analysis comment",
        "expressivenessComment": "expressiveness comment"
      },
      "vote": 1,
      "comments": [
        {
          "id": 60,
          "createdBy": {
            "id": 10,
            "email": "test",
            "name": "tedyeee",
            "reputation": 16
          },
          "comment": "this is comment.",
          "createdAt": "2017-10-09T10:49:26.719"
        }
      ],
      "createdBy": {
        "id": 10,
        "email": "test",
        "name": "tedyeee",
        "reputation": 16
      },
      "createdAt": "2017-10-09T10:48:22.298"
    }
  ]
}
