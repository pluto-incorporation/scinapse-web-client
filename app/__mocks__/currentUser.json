{
  "affiliation": "eeee",
  "articleCount": 0,
  "email": "honeyjasdds@postech.ac.kr",
  "emailVerified": false,
  "id": "39",
  "isLoggedIn": true,
  "major": null,
  "name": "academey",
  "oauth": null,
  "oauthLoggedIn": true,
  "profileImage": null,
  "reputation": 0,
  "reviewCount": 0
}
