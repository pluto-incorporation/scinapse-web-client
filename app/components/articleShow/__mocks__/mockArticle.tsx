export function getMockArticle() {
  return {
    articleId: 1,
    tags: ["Open Access Paper", "CAU Paper"],
  };
}
