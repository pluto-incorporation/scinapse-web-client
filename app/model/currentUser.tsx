import { Member } from "./member";

export interface CurrentUser
  extends Member,
    Readonly<{
      isLoggedIn: boolean;
      oauthLoggedIn: boolean;
    }> {}

export const CURRENT_USER_INITIAL_STATE: CurrentUser = {
  isLoggedIn: false,
  oauthLoggedIn: false,
  email: "",
  firstName: "",
  lastName: "",
  id: 0,
  profileImage: "",
  profileId: null,
  affiliation: "",
  major: "",
  commentCount: 0,
  emailVerified: false,
  oauth: null,
};
