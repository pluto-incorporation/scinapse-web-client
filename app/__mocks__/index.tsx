import { IArticle, recordifyArticle } from "../model/article";
import { IWallet, WalletFactory } from "../model/wallet";
import { IAuthor, recordifyAuthor } from "../model/author";
import { IMember, recordifyMember } from "../model/member";
import { IArticlePoint, ArticlePointFactory } from "../model/articlePoint";
import { IComment, recordifyComment } from "../model/comment";
import { IReviewPoint, ReviewPointFactory } from "../model/reviewPoint";
import { IReview, recordifyReview } from "../model/review";

export const RAW = {
  ARTICLE: require("./article.json") as IArticle,
  ARTICLE_POINT: require("./articlePoint.json") as IArticlePoint,
  AUTHOR: require("./author.json") as IAuthor,
  COMMENT: require("./comment.json") as IComment,
  REVIEW: require("./review.json") as IReview,
  REVIEW_POINT: require("./reviewPoint.json") as IReviewPoint,
  MEMBER: require("./member.json") as IMember,
  WALLET: require("./wallet.json") as IWallet,
};

export const RECORD = {
  ARTICLE: recordifyArticle(RAW.ARTICLE),
  ARTICLE_POINT: ArticlePointFactory(RAW.ARTICLE_POINT),
  AUTHOR: recordifyAuthor(RAW.AUTHOR),
  COMMENT: recordifyComment(RAW.COMMENT),
  REVIEW: recordifyReview(RAW.REVIEW),
  REVIEW_POINT: ReviewPointFactory(RAW.REVIEW_POINT),
  MEMBER: recordifyMember(RAW.MEMBER),
  WALLET: WalletFactory(RAW.WALLET),
};
