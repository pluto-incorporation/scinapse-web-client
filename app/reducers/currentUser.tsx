import { IReduxAction } from "../typings/actionType";
import { ACTION_TYPES } from "../actions/actionTypes";
import { CURRENT_USER_INITIAL_STATE, ICurrentUserRecord, recordifyCurrentUser } from "../model/currentUser";

export function reducer(state = CURRENT_USER_INITIAL_STATE, action: IReduxAction<any>): ICurrentUserRecord {
  switch (action.type) {
    case ACTION_TYPES.SIGN_IN_SUCCEEDED_TO_SIGN_IN: {
      return recordifyCurrentUser(action.payload.user).withMutations(currentUser => {
        currentUser.set("isLoggedIn", action.payload.loggedIn).set("oauthLoggedIn", action.payload.oauthLoggedIn);
      });
    }

    case ACTION_TYPES.AUTH_SUCCEEDED_TO_SIGN_OUT: {
      return CURRENT_USER_INITIAL_STATE;
    }

    case ACTION_TYPES.AUTH_SUCCEEDED_TO_CHECK_LOGGED_IN: {
      return recordifyCurrentUser(action.payload.user).withMutations(currentUser => {
        currentUser.set("isLoggedIn", action.payload.loggedIn).set("oauthLoggedIn", action.payload.oauthLoggedIn);
      });
    }

    case ACTION_TYPES.PROFILE_SUCCEEDED_TO_UPDATE_USER_PROFILE: {
      return recordifyCurrentUser(action.payload.userProfile).set("isLoggedIn", true);
    }

    case ACTION_TYPES.EMAIL_VERIFICATION_SUCCEEDED_TO_VERIFY_TOKEN: {
      return state.set("emailVerified", true);
    }

    default:
      return state;
  }
}
