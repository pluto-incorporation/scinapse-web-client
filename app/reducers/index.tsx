import * as Redux from "redux";
import { RouterState } from "connected-react-router";
import * as ConfigurationReducer from "./configuration";
import * as currentUserReducer from "./currentUser";
import * as signUpReducer from "../components/auth/signUp/reducer";
import * as signInReducer from "../components/auth/signIn/reducer";
import { CURRENT_USER_INITIAL_STATE, CurrentUser } from "../model/currentUser";
import * as dialogReducer from "../components/dialog/reducer";
import * as layoutReducer from "../components/layouts/reducer";
import { LAYOUT_INITIAL_STATE, LayoutState } from "../components/layouts/records";
import * as articleSearchReducer from "../components/articleSearch/reducer";
import { ARTICLE_SEARCH_INITIAL_STATE, ArticleSearchState } from "../components/articleSearch/records";
import * as emailVerificationReducer from "../components/auth/emailVerification/reducer";
import { PaperShowState, PAPER_SHOW_INITIAL_STATE } from "../components/paperShow/records";
import { reducer as paperShowReducer } from "../components/paperShow/reducer";
import {
  reducer as AuthorShowReducer,
  AuthorShowState,
  AUTHOR_SHOW_INITIAL_STATE,
} from "../components/authorShow/reducer";
import { reducer as EntityReducer, INITIAL_ENTITY_STATE, EntityState } from "./entity";
import * as homeReducer from "../components/home/reducer";
import { HomeState, HOME_INITIAL_STATE } from "../components/home/records";
import {
  ProfileNewState,
  PROFILE_NEW_STATE_INITIAL_STATE,
  reducer as ProfileNewReducer,
} from "../containers/newProfile/reducer";
import {
  reducer as CollectionShowReducer,
  CollectionShowState,
  INITIAL_COLLECTION_SHOW_STATE,
} from "../components/collectionShow/reducer";
import {
  reducer as UserCollectionsReducer,
  UserCollectionsState,
  USER_COLLECTIONS_INITIAL_STATE,
} from "../components/collections/reducer";
import {
  reducer as JournalShowReducer,
  JournalShowState,
  JOURNAL_SHOW_INITIAL_STATE,
} from "../components/journalShow/reducer";
import {
  ProfileShowState,
  PROFILE_SHOW_STATE_INITIAL_STATE,
  reducer as ProfileReducer,
} from "../containers/profileShow/reducer";

export interface AppState {
  router?: RouterState;
  configuration: ConfigurationReducer.Configuration;
  signUp: signUpReducer.SignUpState;
  signIn: signInReducer.SignInState;
  dialog: dialogReducer.DialogState;
  layout: LayoutState;
  home: HomeState;
  emailVerification: emailVerificationReducer.EmailVerificationState;
  currentUser: CurrentUser | null;
  articleSearch: ArticleSearchState;
  paperShow: PaperShowState;
  authorShow: AuthorShowState;
  journalShow: JournalShowState;
  collectionShow: CollectionShowState;
  userCollections: UserCollectionsState;
  profileShow: ProfileShowState;
  profileNew: ProfileNewState;
  entities: EntityState;
}

export const initialState: AppState = {
  configuration: ConfigurationReducer.CONFIGURATION_INITIAL_STATE,
  signUp: signUpReducer.SIGN_UP_INITIAL_STATE,
  signIn: signInReducer.SIGN_IN_INITIAL_STATE,
  dialog: dialogReducer.DIALOG_INITIAL_STATE,
  home: HOME_INITIAL_STATE,
  layout: LAYOUT_INITIAL_STATE,
  emailVerification: emailVerificationReducer.EMAIL_VERIFICATION_INITIAL_STATE,
  currentUser: CURRENT_USER_INITIAL_STATE,
  articleSearch: ARTICLE_SEARCH_INITIAL_STATE,
  paperShow: PAPER_SHOW_INITIAL_STATE,
  authorShow: AUTHOR_SHOW_INITIAL_STATE,
  journalShow: JOURNAL_SHOW_INITIAL_STATE,
  collectionShow: INITIAL_COLLECTION_SHOW_STATE,
  userCollections: USER_COLLECTIONS_INITIAL_STATE,
  profileShow: PROFILE_SHOW_STATE_INITIAL_STATE,
  profileNew: PROFILE_NEW_STATE_INITIAL_STATE,
  entities: INITIAL_ENTITY_STATE,
};

export const rootReducer: Redux.Reducer<AppState> = Redux.combineReducers({
  configuration: ConfigurationReducer.reducer,
  signUp: signUpReducer.reducer,
  signIn: signInReducer.reducer,
  dialog: dialogReducer.reducer,
  home: homeReducer.reducer,
  layout: layoutReducer.reducer,
  articleSearch: articleSearchReducer.reducer,
  emailVerification: emailVerificationReducer.reducer,
  paperShow: paperShowReducer,
  authorShow: AuthorShowReducer,
  journalShow: JournalShowReducer,
  currentUser: currentUserReducer.reducer,
  collectionShow: CollectionShowReducer,
  userCollections: UserCollectionsReducer,
  profileShow: ProfileReducer,
  profileNew: ProfileNewReducer,
  entities: EntityReducer,
});
