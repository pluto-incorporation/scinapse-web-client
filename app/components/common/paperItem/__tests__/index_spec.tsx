import * as React from "react";
import * as renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import PaperItem from "..";
import { RAW } from "../../../../__mocks__";
import { CURRENT_USER_INITIAL_STATE } from "../../../../model/currentUser";
import { generateMockStore } from "../../../../__tests__/mockStore";
import { initialState } from "../../../../reducers";

describe("PaperItem Component", () => {
  const mockStore = generateMockStore(initialState);
  describe("when paper data has DOI", () => {
    it("should render correctly", () => {
      const tree = renderer
        .create(
          <Provider store={mockStore}>
            <MemoryRouter>
              <PaperItem pageType="paperShow" paper={RAW.PAPER} currentUser={CURRENT_USER_INITIAL_STATE} />
            </MemoryRouter>
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe("when paper data doesn't have DOI", () => {
    it("should render correctly", () => {
      const mockPaper = { ...RAW.PAPER, doi: "" };
      const tree = renderer
        .create(
          <Provider store={mockStore}>
            <MemoryRouter>
              <PaperItem pageType="paperShow" paper={mockPaper} currentUser={CURRENT_USER_INITIAL_STATE} />
            </MemoryRouter>
          </Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
