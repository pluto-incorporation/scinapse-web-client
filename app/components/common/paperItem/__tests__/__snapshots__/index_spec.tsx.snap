// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaperItem Component when paper data doesn't have DOI should render correctly 1`] = `
<div
  className="paperItemWrapper"
>
  <div
    className="contentSection"
  >
    <div
      className="itemWrapper"
    >
      <i
        className="icon bookmarkIcon"
      />
      <span
        className="subContext"
      >
        Saved in 
      </span>
      <a
        className="collectionTitle"
        href="/collections/22"
        onClick={[Function]}
      >
        Read Later
      </a>
    </div>
    <div>
      <a
        className="title"
        href="/papers/101"
        onClick={[Function]}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Preparation and characterization of <b>graphene</b> oxide paper.",
            }
          }
        />
      </a>
      <a
        className="newTabIconWrapper"
        href="/papers/101"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        <i
          className="icon newTabIcon"
        />
      </a>
    </div>
    <div
      className="publishInfoList"
    >
      <div
        className="venue"
      >
        <i
          className="icon"
        />
        <div
          className="journalText"
        >
          <span
            className="bold"
          >
            2007
             in 
          </span>
          <a
            className="venueName"
            href="/journals/10"
            onClick={[Function]}
          >
            Nature
          </a>
          <span
            className="bold"
          >
             [IF: 40.14]
          </span>
        </div>
      </div>
      <div
        className="author"
      >
        <i
          className="icon"
        />
        <span
          className="authors"
        >
          <span
            className="author"
          >
            <a
              className="authorName"
              href="/authors/undefined"
              onClick={[Function]}
            >
              Dmitriy A. Dikin
            </a>
             
            <span>
              , 
            </span>
          </span>
          <span
            className="author"
          >
            <a
              className="authorName"
              href="/authors/undefined"
              onClick={[Function]}
            >
              Sasha Stankovich
            </a>
             
          </span>
          <span
            className="toggleAuthorsButton"
            onClick={[Function]}
          >
            + 5 Authors
          </span>
          <span
            className="author"
          >
            <a
              className="authorName"
              href="/authors/undefined"
              onClick={[Function]}
            >
              Rodney S. Ruoff
            </a>
             
          </span>
        </span>
      </div>
    </div>
    <span
      className="abstract"
      style={Object {}}
    >
      <span>
        Paper- or foil-like materials are used in many technological applications, as protective layers, chemical filters and adhesives, for instance. So-called 'bucky' paper, based on carbon nanotubes, has the potential to extend the use of high-tech papers to applications such as fuel cells, as it has good mechanical and electrical properties. Now a new carbon nanocomposite paper-like material has been prepared that promises superior performance in stiffness, strength and flexibility: &lt;b&gt;graphene&lt;/b&gt; ...
      </span>
      <label
        className="moreOrLess"
        onClick={[Function]}
      >
        <span>
          more
        </span>
      </label>
    </span>
    <div
      className="infoList"
    >
      <a
        className="citedButton"
        href="/papers/101#cited"
        onClick={[Function]}
      >
        <span>
          2,909 Citations
        </span>
      </a>
      <a
        className="sourceButton"
        href="https://www.ncbi.nlm.nih.gov/pubmed/17653188"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        <i
          className="icon pdfIconWrapper"
        />
        <span>
          Download PDF
        </span>
      </a>
      <button
        className="addCollectionBtnWrapper"
        onClick={[Function]}
      >
        <i
          className="icon plusIcon"
        />
        <span>
          Add To Collection
        </span>
      </button>
      <div
        className="claimButton"
      >
        <div>
          <button
            className="MuiButtonBase-root-7 MuiIconButton-root-1 additionalMenuIcon"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-6"
            >
              <i
                className="icon ellipsisIcon"
              />
            </span>
            <span
              className="MuiTouchRipple-root-10"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PaperItem Component when paper data has DOI should render correctly 1`] = `
<div
  className="paperItemWrapper"
>
  <div
    className="contentSection"
  >
    <div
      className="itemWrapper"
    >
      <i
        className="icon bookmarkIcon"
      />
      <span
        className="subContext"
      >
        Saved in 
      </span>
      <a
        className="collectionTitle"
        href="/collections/22"
        onClick={[Function]}
      >
        Read Later
      </a>
    </div>
    <div>
      <a
        className="title"
        href="/papers/101"
        onClick={[Function]}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "Preparation and characterization of <b>graphene</b> oxide paper.",
            }
          }
        />
      </a>
      <a
        className="newTabIconWrapper"
        href="/papers/101"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        <i
          className="icon newTabIcon"
        />
      </a>
    </div>
    <div
      className="publishInfoList"
    >
      <div
        className="venue"
      >
        <i
          className="icon"
        />
        <div
          className="journalText"
        >
          <span
            className="bold"
          >
            2007
             in 
          </span>
          <a
            className="venueName"
            href="/journals/10"
            onClick={[Function]}
          >
            Nature
          </a>
          <span
            className="bold"
          >
             [IF: 40.14]
          </span>
        </div>
      </div>
      <div
        className="author"
      >
        <i
          className="icon"
        />
        <span
          className="authors"
        >
          <span
            className="author"
          >
            <a
              className="authorName"
              href="/authors/undefined"
              onClick={[Function]}
            >
              Dmitriy A. Dikin
            </a>
             
            <span>
              , 
            </span>
          </span>
          <span
            className="author"
          >
            <a
              className="authorName"
              href="/authors/undefined"
              onClick={[Function]}
            >
              Sasha Stankovich
            </a>
             
          </span>
          <span
            className="toggleAuthorsButton"
            onClick={[Function]}
          >
            + 5 Authors
          </span>
          <span
            className="author"
          >
            <a
              className="authorName"
              href="/authors/undefined"
              onClick={[Function]}
            >
              Rodney S. Ruoff
            </a>
             
          </span>
        </span>
      </div>
    </div>
    <span
      className="abstract"
      style={Object {}}
    >
      <span>
        Paper- or foil-like materials are used in many technological applications, as protective layers, chemical filters and adhesives, for instance. So-called 'bucky' paper, based on carbon nanotubes, has the potential to extend the use of high-tech papers to applications such as fuel cells, as it has good mechanical and electrical properties. Now a new carbon nanocomposite paper-like material has been prepared that promises superior performance in stiffness, strength and flexibility: &lt;b&gt;graphene&lt;/b&gt; ...
      </span>
      <label
        className="moreOrLess"
        onClick={[Function]}
      >
        <span>
          more
        </span>
      </label>
    </span>
    <div
      className="infoList"
    >
      <a
        className="citedButton"
        href="/papers/101#cited"
        onClick={[Function]}
      >
        <span>
          2,909 Citations
        </span>
      </a>
      <a
        className="sourceButton"
        href="https://www.ncbi.nlm.nih.gov/pubmed/17653188"
        onClick={[Function]}
        rel="noopener"
        target="_blank"
      >
        <i
          className="icon pdfIconWrapper"
        />
        <span>
          Download PDF
        </span>
      </a>
      <span
        className="DOIMetaButtonsWrapper"
      >
        <span
          className="citationIconWrapper"
          onClick={[Function]}
        >
          <i
            className="icon citationIcon"
          />
          <span>
            Cite
          </span>
        </span>
      </span>
      <button
        className="addCollectionBtnWrapper"
        onClick={[Function]}
      >
        <i
          className="icon plusIcon"
        />
        <span>
          Add To Collection
        </span>
      </button>
      <div
        className="claimButton"
      >
        <div>
          <button
            className="MuiButtonBase-root-7 MuiIconButton-root-1 additionalMenuIcon"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-6"
            >
              <i
                className="icon ellipsisIcon"
              />
            </span>
            <span
              className="MuiTouchRipple-root-10"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
